csrc/executor_utils.cpp

```diff
@@ -705,7 +705,18 @@ ExpressionEvaluator bindInputs(
     // NOTE: we bind all inputs here, including at::Tensors. This means that
     // expr_eval will create a PolymorphicValue containing *args[i], which means
     // that at::Tensor's lifetime will be at least as long as that of expr_eval.
-    expr_eval.bind(inputs[i], *args[i], true);
+    try {
+      expr_eval.bind(inputs[i], *args[i], true);
+    } catch (const nvfError& e) {
+      std::stringstream ss;
+      ss << "When trying to run the provided host program,"
+         << " there was an error with the provided input " << i
+         << ". Provided input was:\n  ";
+      ss << PolymorphicValue_functions::toString(*args[i]);
+      ss << "\n  which does not match the expected input:\n  ";
+      ss << inputs[i]->toString() << "\n";
```

Comment:
In this instance I didn't think it was particularly helpful, as it's the caller that we would typically want to point at, not the inside of expression evaluator which is the first to find it. I think we'd actually want the error to be thrown higher where `bindInputs` is called as I expect most instances that would fail are because of providing bad inputs.

WDYT?

Reply:
I trust your judgement -- I haven't ran into enough errors to have a strong opinion. 

AFAICT, `ExpressionEvaluator::bind` is a deep function that can fail at https://github.com/NVIDIA/Fuser/blob/346e51c285f2d511cc2a9455d46dd0c97f5787d2/csrc/evaluator_common.cpp#L369. While I agree with you that "most instances that would fail are because of providing bad inputs", why an input is bad might not be immediately clear to the user.

Reply:
I agree, that's why I lifted the error in the common place we bind a bunch of inputs to expression evaluator provided from someplace not generated by nvFuser (developers in tests and Thunder in integration).